concrete/controllers/dialog/type/update_from_type.php

```diff
@@ -0,0 +1,284 @@
+        foreach ($asociatedBlocks as $asociatedBlock) {
+
+          $bID = $asociatedBlock['bID'];
+          $cbRelationID = $asociatedBlock['cbRelationID'];
+
+          $blockToUpdate = array(
+            'bID' => $bID,
+            'arHandle' => $asociatedBlock['arHandle'],
+            'actions' => array()
+          );
+
+          $blockPageTypeDefaultPage = NULL;
+
+          if(array_key_exists($bID, $pageTypeDefaultPageBlocks)){
+            $blockPageTypeDefaultPage = $pageTypeDefaultPageBlocks[$bID];
+
+          }else if(array_key_exists($cbRelationID, $pageTypeDefaultPageRelBlocks)){
+            $blockPageTypeDefaultPage = $pageTypeDefaultPageRelBlocks[$cbRelationID];
+            $blockToUpdate['actions'][] = array('name' => 'update_forked', 'pageTypeBlockID' => $blockPageTypeDefaultPage->getBlockID(), 'pageTypeArHandle' => $blockPageTypeDefaultPage->getAreaHandle());
+          }else{
+            $blockToUpdate['actions'][] = array('name' => 'delete');
+          }
+
+          if($blockPageTypeDefaultPage){
+            $forceChangeOrder = false;
+
+            if($asociatedBlock['arHandle'] != $blockPageTypeDefaultPage->getAreaHandle()){
+              $blockToUpdate['actions'][] = array(
+                'name' => 'change_arHandle',
+                'actualArHandle' => $blockPageTypeDefaultPage->getAreaHandle()
+              );
+              
+              $forceChangeOrder = true;
+            }
+
+            if($forceChangeOrder || ($asociatedBlock['cbDisplayOrder'] != $blockPageTypeDefaultPage->cbDisplayOrder)){
+              $blockToUpdate['actions'][] = array(
+                'name' => 'change_display_order',
+                'actualDisplayOrder' => $blockPageTypeDefaultPage->cbDisplayOrder,
+                'actualArHandle' => $blockPageTypeDefaultPage->getAreaHandle()
+              );
+            }
+          }
+
+          unset($pageTypeDefaultPageBlocksClone[$bID]);
+
+          if(!empty($blockToUpdate['actions'])){
+            $blocksToUpdate[] = $blockToUpdate;
+          }
+        }
+
+        if(count($pageTypeDefaultPageBlocksClone) > 0){
+          foreach ($pageTypeDefaultPageBlocksClone as $pageTypeBlock) {
+            $blocksToAdd[] = array('bID' => $pageTypeBlock->getBlockID(), 'actualDisplayOrder' => $pageTypeBlock->cbDisplayOrder, 'pageTypeArHandle' => $pageTypeBlock->getAreaHandle());
+          }
+        }
+
+        $records[] = array(
+          'blocksToAdd' => $blocksToAdd,
+          'blocksToUpdate' => $blocksToUpdate,
+          'cID' => $pagePlusCV['cID'],
+          'cvID' => $pagePlusCV['cvID']
+        );
+
+      }
+
+      $queueName = $queue->getName();
+      $queue->deleteQueue();
+      $queue = Queue::get($queueName);
+
+      foreach ($records as $record) {
+          $queue->send(serialize($record));
+      }
+
+      return $queue;
+    }
+
+    public function submit($ptID, $pTemplateID)
+    {
+
+        $responseFactory = $this->app->make(ResponseFactory::class);
```

Comment:
@mlocati ok, thanks for the info. I'm a little bit mad, because this P.R. has been postponed because of "nice to have" features for months. And the code I took examples of and is currently in production has all issues Your pointing out here... Will correct anyways!